lib/serve_html_task.rb

```diff
@@ -16,13 +16,17 @@
 # you may find current contact information at www.suse.com
 
 class ServeHtmlTask
-  def serve(system_description_store, opts)
-    hostname = Socket.gethostbyname(Socket.gethostname).first
-    url = "http://#{hostname}:#{opts[:port]}/"
-    if opts[:public]
-      remote_ip = Socket.ip_address_list.find { |ip| ip.ipv4? && !ip.ipv4_loopback? }.ip_address
-      remote_url = "http://#{remote_ip}:#{opts[:port]}/"
+  def assemble_url(opts)
+    host = if opts[:public]
+      Socket.gethostname
```

Comment:
Wouldn't it still make sense to try to get the "real" hostname and only fall back to the alias if it fails? I think it's not very unrealistic that the locally configured hostname is not the one under which the machine is reachable from the outside.